packages/mui-joy/src/Paper/Paper.tsx

```diff
@@ -0,0 +1,168 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { alpha } from '@mui/system';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import clsx from 'clsx';
+import { Mode } from 'packages/mui-system/src/cssVars/useCurrentColorScheme';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useColorScheme, useThemeProps } from '../styles';
+import styled from '../styles/styled';
+import { getPaperUtilityClass } from './paperClasses';
+import { PaperProps, PaperTypeMap } from './PaperProps';
+
+// Originally from Material Paper
+// Tweaked to be used for Joy Paper
+const getOverlayAlpha = (elevationProp: 'xs' | 'sm' | 'md' | 'lg' | 'xl') => {
```

Comment:
I think we can remove this fn and rethink how to approach overlay in dark mode again (not in this PR).

- https://github.com/mui/material-ui/issues/29643
- https://github.com/mui/material-ui/issues/28408
- https://github.com/mui/material-ui/issues/27980